src/components/Card.js

```diff
@@ -1,31 +1,45 @@
 export class Card {
-    constructor(data, cardSelector, handleImageClick) {
+    constructor(data, cardSelector, handleImageClick, deleteHandler, 
+            likeHandler, unlikeHandler) {
         this._data = data;
         this._cardSelector = cardSelector;
         this._handleImageClick = handleImageClick;
+        this._deleteHandler = deleteHandler;
+        this._likeHandler = likeHandler;
+        this._unlikeHandler = unlikeHandler;
     }
 
     _setEventListeners() {
         this._element.querySelector(".card__like-button")
-            .addEventListener("click", this._likeHandler);
+            .addEventListener("click", () => { this._likeCard() });
         this._element.querySelector(".card__delete-button")
             .addEventListener("click", this._deleteHandler);
         this._element.querySelector(".card__image")
             .addEventListener("click", this._handleImageClick);
     }
 
-    _deleteHandler(event) {
-        event.target.closest(".card").remove();
+    _likeCard() {
+        if (this._data.isLiked) {
+            this._unlikeHandler(this._data._id)
+                .then(() => {
+                    this._toggleLike();
```

Comment:
Please, don't add `then` inside `Card` because now you don't wait till the server request is fullfilled


You had the correct code last time in `index.js`


```js

  _likeCard() {
    if (this._data.isLiked) {
      this._unlikeHandler(this._data._id);
    } else {
      this._likeHandler(this._data._id);
    }
  }


```


And in `index.js` it should be something like that



```js
async function unlikeCard(card) {
  api
    .unlikeCard(card.id)
    .then(() => {
      card.toggleLike();
    })
    .catch((err) => {
      console.log("Error liking a card: " + err);
    });
}

```


And `toggleLike` should not have `_`